### PR TITLE
Removed warnings for make command

### PR DESCRIPTION
**Pull request by kenwoon**
- Fixed `CYPHERSCAN_H` typo in cypherscan.h
- Fixed passing incompatible pointer types for `psql_scan_cypher_command()` function called in mainloop.c
- Suppressed warnings from unused functions `yyunput` and `input` in cypher.c
- Suppressed warning of 23 useless nonterminals and 40 useless rules in cypher.y
- Removed `printf` statements from the rules in cypher.y
- Modified error message when parsing cypher queries to match that from the error message used in postgresql

### CODE REVIEW

File: cypher.y [hunk 1]
```diff
@@ -37,137 +37,148 @@ int order_clause_direction = 1; // 1 for ascending, -1 for descending
...
     match_clause
-    where_clause_opt
-    with_clause_opt
-    return_clause
+    | where_clause
+    | with_clause
+    | return_clause
     ;
 
 match_clause:
```
Comment (ibrarahmad) [inline cypher.y:60]: We don't need printf anymore

File: cypher.y [hunk 2]
```diff
@@ -37,137 +37,148 @@ int order_clause_direction = 1; // 1 for ascending, -1 for descending
...
+	/* empty */ { printf("No LIMIT clause.\n"); }
+	| LIMIT INTEGER { printf("LIMIT clause parsed: %d.\n", $2); }
+	;
 
 %%
 
 void yyerror(char const *s)
 {
-    fprintf(stderr, "Parser error: %s\n", s);
+	printf("Parser error: %s\n", s);
```
Comment (ibrarahmad) [inline cypher.y]: Change message,should look like postgresql error

**Review by ibrarahmad — CHANGES_REQUESTED** [review]

File: cypher.y (hunk 1 above)
Comment (kenwoon) [reply cypher.y:60]: Fixed!

File: cypher.y (hunk 2 above)
Comment (kenwoon) [reply cypher.y]: Fixed!

File: cypher.y [hunk 3]
```diff
@@ -55,130 +55,129 @@ query:
...
+	/* empty */
+	| LIMIT INTEGER
 	;
 
 %%
 
 void yyerror(char const *s)
 {
-	printf("Parser error: %s\n", s);
+	printf("postgresql error: %s\n", s);
```
Comment (ibrarahmad) [inline cypher.y]: ERROR:  syntax error at or near "SER"

**Review by ibrarahmad — CHANGES_REQUESTED** [review]

File: cypher.y (hunk 3 above)
Comment (kenwoon) [reply cypher.y]: Fixed!

File: cypher.y [hunk 4]
```diff
@@ -37,30 +40,30 @@ int order_clause_direction = 1; // 1 for ascending, -1 for descending
 %token UNKNOWN
 
 %left PIPE
-%left ARROW
+%left ARROWi
```
Comment (ibrarahmad) [inline cypher.y]: What is that?

File: cypher.y [hunk 5]
```diff
@@ -79,15 +81,18 @@ node_properties_opt:
...
 rel_type:
-    /* no action needed */
+    COLON IDENTIFIER
+     | LBRACKET IDENTIFIER RBRACKET
     ;
 
 rel_direction:
-    /* no action needed */
+    ARROW
+//    | ARROW rel_type_name ARROW
```
Comment (ibrarahmad) [inline cypher.y]: Don't use c++ style comments.

**Review by ibrarahmad — CHANGES_REQUESTED** [review]

File: cypher.y (hunk 4 above)
Comment (kenwoon) [reply cypher.y]: Fixed the typo

File: cypher.y (hunk 5 above)
Comment (kenwoon) [reply cypher.y]: Removed `//` and replaced `rel_type_name` with `IDENTIFIER`

**Pull request merged**